### PR TITLE
feat(notification-settings-screen): UX & navigation & redux flow

### PR DESCRIPTION
**Pull request by VladislavRiabko**
- add isNotificationAllow to state 
- add setNotificationAllow action 
- add ExtendedSwitch component 
- add MenuSwitch component 
- add NotificationSettings screen UX

### CODE REVIEW

File: app/redux/slices/settingsSlice.ts [hunk 1]
```diff
@@ -1,21 +1,31 @@
...
-    setLanguage(state, { payload: { language } }: ISettingsActionPayload) {
+    setLanguage(state, { payload: { language } }: ILanguageActionPayload) {
       state.language = language;
     },
+    setNotificationAllow(
+      state,
+      { payload: { isNotificationAllow } }: INotificationAllowActionPayload,
+    ) {
+      state.isNotificationAllow = isNotificationAllow;
+    },
```
Comment (emberglaz) [inline app/redux/slices/settingsSlice.ts]: refactor

File: app/redux/types/settingsTypes.ts [hunk 2]
```diff
@@ -1,5 +1,9 @@
 import { PayloadAction } from '@reduxjs/toolkit';
 
-export type ISettingsActionPayload = PayloadAction<{
+export type ILanguageActionPayload = PayloadAction<{
   language: string;
 }>;
+
+export type INotificationAllowActionPayload = PayloadAction<{
+  isNotificationAllow: boolean;
+}>;
```
Comment (emberglaz) [inline app/redux/types/settingsTypes.ts]: the same

File: app/redux/slices/settingsSlice.ts [hunk 3]
```diff
@@ -1,21 +1,31 @@
...
-
-import { ISettingsActionPayload } from '../types';
+import {
+  ILanguageActionPayload,
+  INotificationAllowActionPayload,
+} from '../types';
 
 interface ISettingsState {
   language: string;
+  isNotificationAllow: boolean;
```
Comment (Reykjavik151) [inline app/redux/slices/settingsSlice.ts]: rename to `isNotificationsEnabled`
all settings options should be with suffix `Enabled` for switches

File: app/redux/types/settingsTypes.ts [hunk 4]
```diff
@@ -1,5 +1,9 @@
 import { PayloadAction } from '@reduxjs/toolkit';
 
-export type ISettingsActionPayload = PayloadAction<{
+export type ILanguageActionPayload = PayloadAction<{
   language: string;
 }>;
+
+export type INotificationAllowActionPayload = PayloadAction<{
+  isNotificationAllow: boolean;
```
Comment (Reykjavik151) [inline app/redux/types/settingsTypes.ts]: The same as above

File: app/redux/types/settingsTypes.ts [hunk 5]
```diff
@@ -1,5 +1,9 @@
 import { PayloadAction } from '@reduxjs/toolkit';
 
-export type ISettingsActionPayload = PayloadAction<{
+export type ILanguageActionPayload = PayloadAction<{
   language: string;
 }>;
+
+export type INotificationAllowActionPayload = PayloadAction<{
```
Comment (Reykjavik151) [inline app/redux/types/settingsTypes.ts]: rename to `ChangeNotificationSetting`

File: app/screens/Menu/NotificationSettings/NotificationSettings.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+export const NotificationSettingsScreen: React.FC<INotificationSettingsScreenProps> =
+  () => {
+    const { t } = useTranslation();
+
+    const dispatch = useAppDispatch();
+
+    const [isNotificationAllow, setIsNotificationAllow] = useState(
+      useAppSelector(state => state.settings.isNotificationAllow),
+    );
```
Comment (Reykjavik151) [inline app/screens/Menu/NotificationSettings/NotificationSettings.tsx]: the same as above

File: app/screens/Menu/NotificationSettings/NotificationSettings.tsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+    const setNotificationAllow = useCallback(() => {
+      setIsNotificationAllow(!isNotificationAllow);
+      dispatch(
+        settingsSlice.actions.setNotificationAllow({
+          isNotificationAllow: !isNotificationAllow,
+        }),
+      );
+    }, [isNotificationAllow, setIsNotificationAllow, dispatch]);
```
Comment (Reykjavik151) [inline app/screens/Menu/NotificationSettings/NotificationSettings.tsx]: the same as above

File: app/screens/Menu/NotificationSettings/NotificationSettings.tsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+          // TODO: change to correct icon
+          rightIcon={IMAGES.WHITE_BACK_ARROW}
+        />
+        <View style={styles.container}>
+          <ExtendedText style={styles.title}>
+            {t('screens.menu.notification_settings.title')}
+          </ExtendedText>
+
+          <MenuSwitch
+            title={t('screens.menu.notification_settings.allow_notifications')}
```
Comment (Reykjavik151) [inline app/screens/Menu/NotificationSettings/NotificationSettings.tsx]: the same as above (naming)

File: app/screens/Menu/components/MenuSwitch/MenuSwitch.tsx [hunk 9]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { View } from 'react-native';
+
+import { ExtendedSwitch, ExtendedText } from '../../../../components';
+import { IMenuSwitchProps } from './MenuSwitch.props';
+import { styles } from './MenuSwitch.styles';
+
+export const MenuSwitch: React.FC<IMenuSwitchProps> = ({
```
Comment (Reykjavik151) [inline app/screens/Menu/components/MenuSwitch/MenuSwitch.tsx]: Maybe name it as `MenuSwitchRow`?

**Review by Reykjavik151 — CHANGES_REQUESTED** [review]

File: app/redux/types/settingsTypes.ts [hunk 10]
```diff
@@ -1,5 +1,8 @@
...
+  language?: string;
+  isNotificationsEnabled?: boolean;
+}
 
-export type ISettingsActionPayload = PayloadAction<{
-  language: string;
-}>;
+export type ISettingsDataPayload = {
+  payload: Partial<ISettingsData>;
+};
```
Comment (Reykjavik151) [inline app/redux/types/settingsTypes.ts]: rename to `setSettings`

**Review by Reykjavik151 — CHANGES_REQUESTED** [review]

File: app/redux/slices/settingsSlice.ts [hunk 11]
```diff
@@ -1,21 +1,26 @@
 import { createSlice } from '@reduxjs/toolkit';
+import { ISetSettingsDataPayload, ISettingsData } from '../types';
 
-import { ISettingsActionPayload } from '../types';
```
Comment (emberglaz) [inline app/redux/slices/settingsSlice.ts]: refactor inputs

**Review by emberglaz — CHANGES_REQUESTED** [review]

**Review by emberglaz — APPROVED** [review]

**Pull request merged**